Deprecate bytecode version 3

iseeyuan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #57775

The minimum supported bytecode version is updated from 3 to 4. We no longer support version 3 bytecode models.

Why?
* There are hacky codes in operator loading, that performs differently on one operator on the global bytecode version 3. Instead operator related metadata should be passed (for example, in #56845). To allow future development, we remove the hacky way first. 
* The bytecode version was bumped from 3 to 4 more than half a year ago. Since all the production models are all bumped to version 4, it's not practical to keep and maintain version 3. The risk to deprecate version 3 is low.

Differential Revision: [D28270791](https://our.internmc.facebook.com/intern/diff/D28270791)

iseeyuan committed:
Deprecate bytecode version 3

[ghstack-poisoned]

iseeyuan committed:
Update on "Deprecate bytecode version 3"


Why?
* There are hacky codes in operator loading, that performs differently on one operator on the global bytecode version 3. Instead operator related metadata should be passed (for example, in #56845). To allow future development, we remove the hacky way first. 
* The bytecode version was bumped from 3 to 4 more than half a year ago. Since all the production models are all bumped to version 4, it's not practical to keep and maintain version 3. The risk to deprecate version 3 is low. 



[ghstack-poisoned]

iseeyuan committed:
Update on "Deprecate bytecode version 3"


The minimum supported bytecode version is updated from 3 to 4. We no longer support version 3 bytecode models.

Why?
* There are hacky codes in operator loading, that performs differently on one operator on the global bytecode version 3. Instead operator related metadata should be passed (for example, in #56845). To allow future development, we remove the hacky way first. 
* The bytecode version was bumped from 3 to 4 more than half a year ago. Since all the production models are all bumped to version 4, it's not practical to keep and maintain version 3. The risk to deprecate version 3 is low. 



[ghstack-poisoned]

iseeyuan commented:
@iseeyuan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28270791).

raziel commented on torch/csrc/jit/mobile/import.cpp:
```diff
@@ -280,7 +278,7 @@ void BytecodeDeserializer::parseMethods(
   // introduced. The old models (some of them already in production) without
-  // version number don't have to be re-generated.
+  // version number are seen as version 3 (deprecated).
   int64_t model_version = 0x3L;
```
I'd suggest making it a constant itself (e.g. default legacy version or something like that).
Is this the only place that will use this constant?
Otherwise maybe move it to caffe2/serialize/versions.h ?

iseeyuan replied:
Good catch. I'll add const here. It's the only place that it's used.

iseeyuan replied:
Good catch. I'll add const here. It's the only place that it's used.

raziel reviewed (approved):
Thanks.
Minor comment on whether to define a constant for the default version.

iseeyuan committed:
Update on "Deprecate bytecode version 3"


The minimum supported bytecode version is updated from 3 to 4. We no longer support version 3 bytecode models.

Why?
* There are hacky codes in operator loading, that performs differently on one operator on the global bytecode version 3. Instead operator related metadata should be passed (for example, in #56845). To allow future development, we remove the hacky way first. 
* The bytecode version was bumped from 3 to 4 more than half a year ago. Since all the production models are all bumped to version 4, it's not practical to keep and maintain version 3. The risk to deprecate version 3 is low.

Differential Revision: [D28270791](https://our.internmc.facebook.com/intern/diff/D28270791)

[ghstack-poisoned]

iseeyuan commented:
@iseeyuan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28270791).

iseeyuan committed:
Update on "Deprecate bytecode version 3"


The minimum supported bytecode version is updated from 3 to 4. We no longer support version 3 bytecode models.

Why?
* There are hacky codes in operator loading, that performs differently on one operator on the global bytecode version 3. Instead operator related metadata should be passed (for example, in #56845). To allow future development, we remove the hacky way first. 
* The bytecode version was bumped from 3 to 4 more than half a year ago. Since all the production models are all bumped to version 4, it's not practical to keep and maintain version 3. The risk to deprecate version 3 is low.

Differential Revision: [D28270791](https://our.internmc.facebook.com/intern/diff/D28270791)

[ghstack-poisoned]

Commit:
Update on "Deprecate bytecode version 3"


The minimum supported bytecode version is updated from 3 to 4. We no longer support version 3 bytecode models.

Why?
* There are hacky codes in operator loading, that performs differently on one operator on the global bytecode version 3. Instead operator related metadata should be passed (for example, in #56845). To allow future development, we remove the hacky way first. 
* The bytecode version was bumped from 3 to 4 more than half a year ago. Since all the production models are all bumped to version 4, it's not practical to keep and maintain version 3. The risk to deprecate version 3 is low.

Differential Revision: [D28270791](https://our.internmc.facebook.com/intern/diff/D28270791)

[ghstack-poisoned]

Commit:
Update on "Deprecate bytecode version 3"


The minimum supported bytecode version is updated from 3 to 4. We no longer support version 3 bytecode models.

Why?
* There are hacky codes in operator loading, that performs differently on one operator on the global bytecode version 3. Instead operator related metadata should be passed (for example, in #56845). To allow future development, we remove the hacky way first. 
* The bytecode version was bumped from 3 to 4 more than half a year ago. Since all the production models are all bumped to version 4, it's not practical to keep and maintain version 3. The risk to deprecate version 3 is low.

Differential Revision: [D28270791](https://our.internmc.facebook.com/intern/diff/D28270791)

[ghstack-poisoned]

Commit:
Update on "Deprecate bytecode version 3"


The minimum supported bytecode version is updated from 3 to 4. We no longer support version 3 bytecode models.

Why?
* There are hacky codes in operator loading, that performs differently on one operator on the global bytecode version 3. Instead operator related metadata should be passed (for example, in #56845). To allow future development, we remove the hacky way first. 
* The bytecode version was bumped from 3 to 4 more than half a year ago. Since all the production models are all bumped to version 4, it's not practical to keep and maintain version 3. The risk to deprecate version 3 is low.

Differential Revision: [D28270791](https://our.internmc.facebook.com/intern/diff/D28270791)

[ghstack-poisoned]

Commit:
Update on "Deprecate bytecode version 3"


The minimum supported bytecode version is updated from 3 to 4. We no longer support version 3 bytecode models.

Why?
* There are hacky codes in operator loading, that performs differently on one operator on the global bytecode version 3. Instead operator related metadata should be passed (for example, in #56845). To allow future development, we remove the hacky way first. 
* The bytecode version was bumped from 3 to 4 more than half a year ago. Since all the production models are all bumped to version 4, it's not practical to keep and maintain version 3. The risk to deprecate version 3 is low.

Differential Revision: [D28270791](https://our.internmc.facebook.com/intern/diff/D28270791)

[ghstack-poisoned]

Commit:
Update on "Deprecate bytecode version 3"


The minimum supported bytecode version is updated from 3 to 4. We no longer support version 3 bytecode models.

Why?
* There are hacky codes in operator loading, that performs differently on one operator on the global bytecode version 3. Instead operator related metadata should be passed (for example, in #56845). To allow future development, we remove the hacky way first. 
* The bytecode version was bumped from 3 to 4 more than half a year ago. Since all the production models are all bumped to version 4, it's not practical to keep and maintain version 3. The risk to deprecate version 3 is low.

Differential Revision: [D28270791](https://our.internmc.facebook.com/intern/diff/D28270791)

[ghstack-poisoned]

Commit:
Update on "Deprecate bytecode version 3"


The minimum supported bytecode version is updated from 3 to 4. We no longer support version 3 bytecode models.

Why?
* There are hacky codes in operator loading, that performs differently on one operator on the global bytecode version 3. Instead operator related metadata should be passed (for example, in #56845). To allow future development, we remove the hacky way first. 
* The bytecode version was bumped from 3 to 4 more than half a year ago. Since all the production models are all bumped to version 4, it's not practical to keep and maintain version 3. The risk to deprecate version 3 is low.

Differential Revision: [D28270791](https://our.internmc.facebook.com/intern/diff/D28270791)

[ghstack-poisoned]

Commit:
Update on "Deprecate bytecode version 3"


The minimum supported bytecode version is updated from 3 to 4. We no longer support version 3 bytecode models.

Why?
* There are hacky codes in operator loading, that performs differently on one operator on the global bytecode version 3. Instead operator related metadata should be passed (for example, in #56845). To allow future development, we remove the hacky way first. 
* The bytecode version was bumped from 3 to 4 more than half a year ago. Since all the production models are all bumped to version 4, it's not practical to keep and maintain version 3. The risk to deprecate version 3 is low.

Differential Revision: [D28270791](https://our.internmc.facebook.com/intern/diff/D28270791)

[ghstack-poisoned]

Commit:
Update on "Deprecate bytecode version 3"


The minimum supported bytecode version is updated from 3 to 4. We no longer support version 3 bytecode models.

Why?
* There are hacky codes in operator loading, that performs differently on one operator on the global bytecode version 3. Instead operator related metadata should be passed (for example, in #56845). To allow future development, we remove the hacky way first. 
* The bytecode version was bumped from 3 to 4 more than half a year ago. Since all the production models are all bumped to version 4, it's not practical to keep and maintain version 3. The risk to deprecate version 3 is low.

Differential Revision: [D28270791](https://our.internmc.facebook.com/intern/diff/D28270791)

[ghstack-poisoned]

Commit:
Update on "Deprecate bytecode version 3"


The minimum supported bytecode version is updated from 3 to 4. We no longer support version 3 bytecode models.

Why?
* There are hacky codes in operator loading, that performs differently on one operator on the global bytecode version 3. Instead operator related metadata should be passed (for example, in #56845). To allow future development, we remove the hacky way first. 
* The bytecode version was bumped from 3 to 4 more than half a year ago. Since all the production models are all bumped to version 4, it's not practical to keep and maintain version 3. The risk to deprecate version 3 is low.

Differential Revision: [D28270791](https://our.internmc.facebook.com/intern/diff/D28270791)

[ghstack-poisoned]
